src/cueUtilities.js

```diff
@@ -319,60 +320,109 @@ module.exports = function (params, cy, api) {
-            var expandcollapseRenderedEndX = expandcollapseRenderedStartX + expandcollapseRenderedRectSize;
-            var expandcollapseRenderedEndY = expandcollapseRenderedStartY + expandcollapseRenderedRectSize;
-                    
-                    var cyRenderedPos = event.renderedPosition || event.cyRenderedPosition;
-            var cyRenderedPosX = cyRenderedPos.x;
-            var cyRenderedPosY = cyRenderedPos.y;
-            var factor = (opts.expandCollapseCueSensitivity - 1) / 2;
-
-            if ( (Math.abs(oldMousePos.x - currMousePos.x) < 5 && Math.abs(oldMousePos.y - currMousePos.y) < 5)
-              && cyRenderedPosX >= expandcollapseRenderedStartX - expandcollapseRenderedRectSize * factor
-              && cyRenderedPosX <= expandcollapseRenderedEndX + expandcollapseRenderedRectSize * factor
-              && cyRenderedPosY >= expandcollapseRenderedStartY - expandcollapseRenderedRectSize * factor
-              && cyRenderedPosY <= expandcollapseRenderedEndY + expandcollapseRenderedRectSize * factor) {
-              if(opts.undoable && !ur)
-                ur = cy.undoRedo({
-                  defaultActions: false
-                });
-              if(api.isCollapsible(node))
-                if (opts.undoable){
-                  ur.do("collapse", {
-                    nodes: node,
-                    options: opts
-                  });
-                }
-                else
-                  api.collapse(node, opts);
-              else if(api.isExpandable(node))
-                if (opts.undoable)
-                  ur.do("expand", {
-                    nodes: node,
-                    options: opts
-                  });
-                else
-                  api.expand(node, opts);
-
-              // don't draw the cue if the group is expandable, as mouse won't be inside it
-              if (hoveredGroup && !isSelectedGroupsContains(hoveredGroup) && api.isExpandable(node)) {
-                hoveredGroup = null;
-              }
-              refreshCanvasImages();
+          return { x: event.changedTouches[0].pageX,
+            y: event.changedTouches[0].pageY
+          };
+        }
+
+        function isEventOnCueRegion(node, event) {
+          // needed so it matches the range for expandcollapses rendered locations
+          const TOUCH_OFFSET_Y = 43;
+          const expandcollapseRenderedStartX = node._private.data.expandcollapseRenderedStartX;
+          let expandcollapseRenderedStartY = node._private.data.expandcollapseRenderedStartY;
+          const expandcollapseRenderedRectSize = node._private.data.expandcollapseRenderedCueSize;
+          const expandcollapseRenderedEndX = expandcollapseRenderedStartX + expandcollapseRenderedRectSize;
+          const expandcollapseRenderedEndY = expandcollapseRenderedStartY + expandcollapseRenderedRectSize;
+
+          let posX;
+          let posY;
+          if (isMouseEvent(event)) {
```

Comment:
Can just check `if (event instanceof MouseEvent)` rather than the function you added. Also, should probably check if `event instanceof TouchEvent` rather than assuming that an event that's not a mouse event must be a touch event.